src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        user.setSalt(HashUtil.salt());
+        String userHashPassword =
+                HashUtil.hashPassword(user.getPassword(), user.getSalt());
+        user.setPassword(userHashPassword);
```

Comment:
The `add` method correctly hashes the user's password with a salt. Ensure that the `HashUtil.salt()` method generates a secure and random salt for each user to maintain security.